### PR TITLE
[suggestion] performance fix for map-matching on core chs

### PR DESCRIPTION
**Pull request by MoKob**
Introduces a pruning weight for the `get_network_distance` function.
This addresses time-outs for map-matching on a core-ch with a large core.

### CODE REVIEW

**Comment by MoKob** [comment]
Things to consider:
We should probably introduce a parameter to select minimal assumed traveling speed.

**Comment by willwhite** [comment]
Looks like this PR should only include dff330f. @danpat did you merge https://github.com/Project-OSRM/osrm-backend/pull/2000 into this branch and push to this branch by accident?

**Comment by willwhite** [comment]
I cleaned up this PR and rebased off of develop.

File: include/engine/plugins/match.hpp [hunk 1]
```diff
@@ -315,7 +315,8 @@ template <class DataFacadeT> class MapMatchingPlugin : public BasePlugin
         SubMatchingList sub_matchings;
         search_engine_ptr->map_matching(candidates_lists, input_coords, input_timestamps,
                                         route_parameters.matching_beta,
-                                        route_parameters.gps_precision, sub_matchings);
+                                        route_parameters.gps_precision, sub_matchings,
+                                        route_parameters.matching_prune_factor );
```
Comment (daniel-j-h) [inline include/engine/plugins/match.hpp]: Format

**Comment by daniel-j-h** [comment]
- [ ] documentation (Wiki?)

**Comment by TheMarex** [comment]
This was added in a different iteration.

**Pull request closed without merging**